--- drivers/media/pci/cx23885/cx23885-dvb.c	2015-04-24 21:44:11.498647000 +0200
+++ drivers/media/pci/cx23885/cx23885-dvb.c	2015-04-24 21:51:47.709848965 +0200
@@ -1894,6 +1894,7 @@
 			si2168_config.i2c_adapter = &adapter;
 			si2168_config.fe = &fe0->dvb.frontend;
 			si2168_config.ts_mode = SI2168_TS_SERIAL;
+			si2168_config.ts_clock_gapped = true;
 			memset(&info, 0, sizeof(struct i2c_board_info));
 			strlcpy(info.type, "si2168", I2C_NAME_SIZE);
 			info.addr = 0x64;
@@ -1940,6 +1941,7 @@
 		si2168_config.i2c_adapter = &adapter;
 		si2168_config.fe = &fe0->dvb.frontend;
 		si2168_config.ts_mode = SI2168_TS_PARALLEL;
+		si2168_config.ts_clock_gapped = true;
 		memset(&info, 0, sizeof(struct i2c_board_info));
 		strlcpy(info.type, "si2168", I2C_NAME_SIZE);
 		info.addr = 0x64;
@@ -2065,11 +2067,13 @@
 		case 1:
 			i2c_bus = &dev->i2c_bus[1];
 			si2168_config.ts_mode = SI2168_TS_PARALLEL;
+			si2168_config.ts_clock_gapped = true;
 			break;
 		/* port c */
 		case 2:
 			i2c_bus = &dev->i2c_bus[0];
 			si2168_config.ts_mode = SI2168_TS_SERIAL;
+			si2168_config.ts_clock_gapped = true;
 			break;
 		}
 
@@ -2155,6 +2159,7 @@
 			si2168_config.i2c_adapter = &adapter;
 			si2168_config.fe = &fe0->dvb.frontend;
 			si2168_config.ts_mode = SI2168_TS_SERIAL;
+			si2168_config.ts_clock_gapped = true;
 			memset(&info, 0, sizeof(struct i2c_board_info));
 			strlcpy(info.type, "si2168", I2C_NAME_SIZE);
 			info.addr = 0x64;
