--- linux/drivers/media/pci/dvbloopback/dvb_loopback.c	2017-12-27 17:47:41.081549015 +0100
+++ linux/drivers/media/pci/dvbloopback/dvb_loopback.c	2017-12-27 17:48:54.646113293 +0100
@@ -1010,7 +1010,7 @@
 
 // create loopback device (will be frontend0)
 	ret = dvb_register_device(&dvblb->adapter, &lbdev->lb_dev,
-	                    &dvbdev_looped, lbdev, type);
+	                    &dvbdev_looped, lbdev, type, 0);
 	if (ret != 0) {
 		info("error registering device adapter%d",
 		     dvblb->adapter.num);
@@ -1018,7 +1018,7 @@
 	}
 // create userspace device (will be frontend1)
 	ret = dvb_register_device(&dvblb->adapter, &lbdev->user_dev,
-	                    &dvbdev_userspace, lbdev, type);
+	                    &dvbdev_userspace, lbdev, type, 0);
 	if (ret != 0) {
 		info("error registering device adapter%d",
 		     dvblb->adapter.num);
