--- drivers/media/pci/Kconfig	2015-03-26 14:00:21.000000000 +0100
+++ drivers/media/pci/Kconfig	2015-04-08 21:25:45.592661996 +0200
@@ -36,6 +36,7 @@
 if MEDIA_DIGITAL_TV_SUPPORT
 	comment "Media digital TV PCI Adapters"
 source "drivers/media/pci/ttpci/Kconfig"
+source "drivers/media/pci/dvbloopback/Kconfig"
 source "drivers/media/pci/b2c2/Kconfig"
 source "drivers/media/pci/pluto2/Kconfig"
 source "drivers/media/pci/dm1105/Kconfig"
--- drivers/media/pci/Makefile	2015-03-26 14:00:21.000000000 +0100
+++ drivers/media/pci/Makefile	2015-04-08 21:26:34.223691236 +0200
@@ -17,6 +17,7 @@
 obj-$(CONFIG_VIDEO_IVTV) += ivtv/
 obj-$(CONFIG_VIDEO_ZORAN) += zoran/
 obj-$(CONFIG_VIDEO_CX18) += cx18/
+obj-$(CONFIG_DVB_LOOPBACK_ADAPTER) += dvbloopback/
 obj-$(CONFIG_VIDEO_CX23885) += cx23885/
 obj-$(CONFIG_VIDEO_CX25821) += cx25821/
 obj-$(CONFIG_VIDEO_CX88) += cx88/
