--- drivers/media/pci/Makefile	2014-11-21 01:17:24.797328571 +0100
+++ drivers/media/pci/Makefile	2014-11-21 01:18:42.027932791 +0100
@@ -21,6 +21,7 @@
 obj-$(CONFIG_VIDEO_CX88) += cx88/
 obj-$(CONFIG_VIDEO_BT848) += bt8xx/
 obj-$(CONFIG_VIDEO_SAA7134) += saa7134/
+obj-$(CONFIG_SAA716X_CORE) += saa716x/
 obj-$(CONFIG_VIDEO_SAA7164) += saa7164/
 obj-$(CONFIG_VIDEO_MEYE) += meye/
 obj-$(CONFIG_STA2X11_VIP) += sta2x11/
