--- drivers/media/dvb-frontends/Makefile	2015-03-18 13:22:50.000000000 +0100
+++ drivers/media/dvb-frontends/Makefile	2015-03-19 10:13:45.751721739 +0100
@@ -77,6 +77,7 @@
 obj-$(CONFIG_DVB_AF9013) += af9013.o
 obj-$(CONFIG_DVB_CX24116) += cx24116.o
 obj-$(CONFIG_DVB_SI21XX) += si21xx.o
+obj-$(CONFIG_DVB_SI2168) += si2168.o
 obj-$(CONFIG_DVB_STV0288) += stv0288.o
 obj-$(CONFIG_DVB_STB6000) += stb6000.o
 obj-$(CONFIG_DVB_S921) += s921.o
--- drivers/media/pci/Makefile	2015-03-18 13:22:50.000000000 +0100
+++ drivers/media/pci/Makefile	2015-03-19 10:11:43.510022390 +0100
@@ -21,6 +21,7 @@
 obj-$(CONFIG_VIDEO_CX88) += cx88/
 obj-$(CONFIG_VIDEO_BT848) += bt8xx/
 obj-$(CONFIG_VIDEO_SAA7134) += saa7134/
+obj-$(CONFIG_SAA716X_CORE) += saa716x/
 obj-$(CONFIG_VIDEO_SAA7164) += saa7164/
 obj-$(CONFIG_VIDEO_MEYE) += meye/
 obj-$(CONFIG_STA2X11_VIP) += sta2x11/
--- drivers/media/tuners/Makefile	2015-03-18 13:22:50.000000000 +0100
+++ drivers/media/tuners/Makefile	2015-03-19 10:15:18.697973232 +0100
@@ -31,6 +31,7 @@
 obj-$(CONFIG_MEDIA_TUNER_E4000) += e4000.o
 obj-$(CONFIG_MEDIA_TUNER_FC2580) += fc2580.o
 obj-$(CONFIG_MEDIA_TUNER_TUA9001) += tua9001.o
+obj-$(CONFIG_MEDIA_TUNER_SI2157) += si2157.o
 obj-$(CONFIG_MEDIA_TUNER_FC0011) += fc0011.o
 obj-$(CONFIG_MEDIA_TUNER_FC0012) += fc0012.o
 obj-$(CONFIG_MEDIA_TUNER_FC0013) += fc0013.o
--- drivers/media/rc/keymaps/Makefile	2015-03-18 13:22:50.000000000 +0100
+++ drivers/media/rc/keymaps/Makefile	2015-03-19 08:04:50.636404461 +0100
@@ -27,6 +27,7 @@
 			rc-dm1105-nec.o \
 			rc-dntv-live-dvb-t.o \
 			rc-dntv-live-dvbt-pro.o \
+			rc-dvbsky.o \
 			rc-em-terratec.o \
 			rc-encore-enltv2.o \
 			rc-encore-enltv.o \
